Trie tree implementation (Java)

ranesr opened the pull request:
This pull request is for issue number #30

ranesr committed:
Trie tree implementation

VamsiSangam commented on Tree Data Structures/Trie Tree/Java/Trie.java:
```diff
@@ -0,0 +1,216 @@
+	}
+	
+	// helper method for searching & startsWith functions
+	public TrieNode searchNode(String word) {
```
Please make the helper method ```private``` and the comment should be saying ```helper method for searching & delete functions``` as the ```startsWith``` function doesn't seem to be using this method.  Please modify the comment.

VamsiSangam commented:
Pretty well written code! Requested minor changes. And before you work on an issue, please comment on the respective issue so that no one else works on the same issue 😄

ranesr committed:
Updated Trie.java

Updated access modifier for searchNode() method & details about it in the comment.

VamsiSangam commented:
Thanks for your contribution! You should be able to see your code live on the [respective post](http://theoryofprogramming.com/2015/01/16/trie-tree-implementation/) in a few minutes.
